src/main/java/uk/gov/hmcts/reform/idam/client/IdamApi.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.idam.client;
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import uk.gov.hmcts.reform.idam.client.models.AuthenticateUserResponse;
+import uk.gov.hmcts.reform.idam.client.models.GeneratePinRequest;
+import uk.gov.hmcts.reform.idam.client.models.GeneratePinResponse;
+import uk.gov.hmcts.reform.idam.client.models.TokenExchangeResponse;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+@FeignClient(name = "idam-api", url = "${idam.api.url}")
```

Comment:
can the url be passed to `IdamClient` constructor instead?

Reply:
Not if we're using the spring integration,
if you're constructing it manually you can, but this is how the functionality works

Reply:
I'd rather let users of this library define a bean. More explicit, but still simple. It's a java library after all, not some spring boot plugin (I assume?)

Reply:
why don't you want to use spring boot? 😉

Reply:
I do, I don't want this library to work only with Spring Boot though and rely of specific configuration

Reply:
it will work without springboot, you can construct it manually.
CCD did it for one of the libraries in their tests as they weren't using the spring context, it works fine